modules/launch-templates/main.tf

```diff
@@ -76,10 +76,14 @@ resource "aws_launch_template" "this" {
     }
   }
 
-  metadata_options {
-    http_endpoint               = each.value.http_endpoint
-    http_tokens                 = each.value.http_tokens
-    http_put_response_hop_limit = each.value.http_put_response_hop_limit
+  dynamic "metadata_options" {
+    for_each = try(each.value.enable_metadata_options, true) ? [1] : []
+
+    content {
+      http_endpoint               = try(each.value.http_endpoint, "enabled")
+      http_tokens                 = try(each.value.http_tokens, "required")
+      http_put_response_hop_limit = try(each.value.http_put_response_hop_limit, 2)
```

Comment:
Should we add [ipv6 support](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#http_protocol_ipv6) as well?